packages/grid/x-grid/src/tests/rendering.XGrid.test.tsx

```diff
@@ -0,0 +1,216 @@
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+import * as React from 'react';
+import {
+  createClientRenderStrictMode,
+  // @ts-expect-error need to migrate helpers to TypeScript
+  waitFor,
+} from 'test/utils';
+import { expect } from 'chai';
+import { XGrid } from '@material-ui/x-grid';
+
+describe('<XGrid /> - Rendering', () => {
+  // TODO v5: replace with createClientRender
+  const render = createClientRenderStrictMode();
+
+  before(function beforeHook() {
+    if (/jsdom/.test(window.navigator.userAgent)) {
+      // Need layouting
+      this.skip();
+    }
+  });
+
+  let rendering: any;
+  let containerSizes: any;
+  let viewportSizes: any;
+  let scrollBar: any;
+
+  function RenderingTest(props) {
+    const { data } = useDemoData({
+      dataSet: 'Commodity',
+      rowLength: 500,
+    });
+
+    return (
+      <div style={{ width: 600, height: 600 }}>
+        <XGrid
+          rows={data.rows}
+          columns={data.columns}
+          onStateChange={(params) => {
+            rendering = params.state.rendering;
+            containerSizes = params.state.containerSizes;
+            viewportSizes = params.state.viewportSizes;
+            scrollBar = params.state.scrollBar;
+          }}
+          {...props}
+        />
+      </div>
+    );
+  }
+
+  beforeEach(() => {
+    rendering = null;
+    containerSizes = null;
+    viewportSizes = null;
+    scrollBar = null;
+  });
+
+  it('should render properly', async () => {
+    render(<RenderingTest />);
+    await waitFor(() => expect(containerSizes).to.not.be.null);
+
+    const expectedRendering = {
```

Comment:
This doesn't seem to be a functional test case, but a snapshot test. A row dump of the state. IMHO, it has no use cases. It will break as soon as we change the slightest state in the data grid, and be a pain to update.

Reply:
Have a look at 1 but I suggest we go for 2 as it's more cost effective for now.
And I think there will be refactoring coming in the rendering engine...

Reply:
OK, sounds great. If we can add dedicated visual regression tests (that fails without the fixes), that would be a great first step.

Reply:
we should be able to as before it would show the scrollbar and now it shouldn't